framework/di/Container.php

```diff
@@ -145,7 +145,8 @@ class Container extends Component
      * In this case, the constructor parameters and object configurations will be used
      * only if the class is instantiated the first time.
      *
-     * @param string|Instance $class the class Instance, name, or an alias name (e.g. `foo`) that was previously
+     * @template T
+     * @param class-string<T>|Instance $class the class Instance, name, or an alias name (e.g. `foo`) that was previously
```

Comment:
Error seems to be because Pslam is more strict and won't let you `new` a generic `class-string` because we don't know what the constructor requires. In this case Yii's DI will handle all that for us so suppressing the Psalm error should be fine and would be handled by other tests/types elsewhere.

Related issue, https://github.com/vimeo/psalm/issues/8628